cookbooks/bcpc-hadoop/recipes/disks.rb

```diff
@@ -89,113 +100,127 @@
     reservation_requests =
       node[:bcpc][:hadoop][:disks][:reservation_requests]
 
-    role_min_disk =
-      node[:bcpc][:hadoop][:disks][:role_min_disk]
-
     #
     # The available disks list is generated dynamically, at converge time.
     #
-    available_disks =
-      node.run_state[:bcpc_hadoop_disks][:available_disks]
-
-    if available_disks.any?
-      available_disks.each_index do |i|
-        Chef::Resource::Directory.new("/disk/#{i}",
-                                      node.run_context).tap do |dd|
-          dd.owner 'root'
-          dd.group 'root'
-          dd.mode 00755
-          dd.recursive true
-          dd.run_action(:create)
-        end
-
-        base_name = available_disks[i]
-
-        #
-        # If the raw disk has already been formatted with an FS, no
-        # partition table is necessary.
-        #
-        # Otherwise, create a partition and format that.
-        #
-        require 'mixlib/shellout'
-
-        fs_check =
-          Mixlib::ShellOut.new('file', '-s', "/dev/#{base_name}")
-        fs_check.run_command
-
-        dev_name = if fs_check.status.success? &&
-                       fs_check.stdout.include?('SGI XFS filesystem')
-                     "/dev/#{base_name}"
-                   else
-                     "/dev/#{base_name}1"
-                   end
-
-        if dev_name.end_with?('1')
-          unless ::File.exist?(dev_name)
-            Chef::Resource::Execute.new("mklabel-#{base_name}",
-                                        node.run_context).tap do |ee|
-              ee.command "parted -s /dev/#{base_name} mklabel gpt"
-              ee.run_action(:run)
-            end
-
-            Chef::Resource::Execute.new("mkpart-#{base_name}",
-                                        node.run_context).tap do |ee|
-              ee.command "parted -s /dev/#{base_name} " \
-                'mkpart bach_data ext4 0% 100%'
-              ee.run_action(:run)
-            end
-          end
-
-        end
+    unformatted_disks =
+      node.run_state[:bcpc_hadoop_disks][:unformatted_disks]
+
+    unformatted_disks.each_index do |i|
+      Chef::Resource::Directory.new("/disk/#{i}",
```

Comment:
I suppose not -- even if it is inserted at the same place in the order, it will not format correctly, because its number will be taken.

This is not a problem for Kafka hosts, running RAID-10.  I hadn't considered its implications on an un-RAIDed node.